### PR TITLE
proposed fix to issue 524

### PR DESCRIPTION
**Pull request by pgkirsch**

### CODE REVIEW

**Comment by whoburg** [comment]
retest this please

**Comment by whoburg** [comment]
I'm going to clean up the multiple pylint commits with a rebase and force-push. @pgkirsch and anyone else with a local copy of the `issue524` branch, please update by doing:
(first, make sure you have no unpushed work, then):
```
git checkout master
git branch -D issue524
git checkout issue524
git pull origin issue524
```

File: gpkit/signomial_program.py [hunk 1]
```diff
@@ -160,8 +160,9 @@ def step(self, x0=None, verbosity=1):
             sp_inits = {vk: vk.descr["sp_init"] for vk in self.negvarkeys
                         if "sp_init" in vk.descr}
             x0.update(sp_inits)
-            # HACK: initial guess for negative variables
-            x0.update({var: 1 for var in self.negvarkeys if var not in x0})
+        # HACK: initial guess for negative variables
+        x0.update({var: 1 for var in self.negvarkeys if var not in x0})
+
```
Comment (whoburg) [inline gpkit/signomial_program.py]: great to see such a simple solution!

File: gpkit/tests/t_sub.py [hunk 2]
```diff
@@ -112,10 +112,20 @@ def test_signomial(self):
...
             self.assertEqual(subbed, 2*x - y - D)
 
+    def test_partial_sub_signomial(self):
+        """Test SP partial x0 initialization"""
+        x = Variable('x')
+        y = Variable('y')
+        with SignomialsEnabled():
+            m = Model(x, [x + y >= 1, y <= 0.5])
+        m.localsolve(x0={x: 0.5}, verbosity=0)
+        self.assertEqual(m.program.gps[0].constraints[0].exp[x], -1./3)
```
Comment (whoburg) [inline gpkit/tests/t_sub.py]: Great to see the test.

not convinced that this test belongs in `t_sub.py` (which is meant for testing `substitution` and `sub` methods -- not initial guesses)

I'd put this in the `TestSP` class in `t_model.py`.

**Comment by whoburg** [comment]
review complete, nice work.

**Comment by galbramc** [comment]
test this please

File: gpkit/tests/t_sub.py (hunk 2 above)
Comment (pgkirsch) [inline gpkit/tests/t_sub.py]: moved this as suggested. If/when checks complete, I will merge the PR.

File: gpkit/signomial_program.py (hunk 1 above)
Comment (bqpd) [inline gpkit/signomial_program.py]: :100:

**Pull request merged**